SPECS/netplan/netplan.spec

```diff
@@ -7,103 +7,241 @@
 Summary:        Network configuration tool using YAML
 Group:          System Environment/Base
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 License:        GPLv3
 URL:            https://netplan.io/
-# Source0:      https://github.com/canonical/%{name}/archive/%{version}/%{version}.tar.gz
-Source0:        %{name}-%{version}.tar.gz
+Source0:        https://github.com/canonical/%{name}/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+
+# netplan build optionally depends on pyflakes, but there is a hard check for it
+# in the meson file. This patch disables that check.
+Patch0:         remove-flakes-check.patch
+
+# Some unit tests parse an openvswitch related config that requires openvswitch
+# to be installed.
+Patch1:         skip-ovs-tests.patch
+
+# Fix bug in netplan when python3-rich is not present.
+Patch2:         rich-import-failure-no-log.patch
+
+# Temporarily disabling broken test suite due to version mismatches between
+# pytest-cov and python3-coverage.
+Patch3:         disable-broken-tests.patch
 
-BuildRequires:  gcc
-BuildRequires:  make
 BuildRequires:  bash-completion-devel
-BuildRequires:  libgcc-devel
 BuildRequires:  bash-devel
-BuildRequires:  systemd-devel
+BuildRequires:  cmake
+BuildRequires:  gcc
 BuildRequires:  glib-devel
+BuildRequires:  libgcc-devel
 BuildRequires:  libyaml-devel
-BuildRequires:  util-linux-devel
+BuildRequires:  meson >= 0.61
+BuildRequires:  ninja-build
 BuildRequires:  python%{python3_pkgversion}-devel
+BuildRequires:  systemd
+BuildRequires:  systemd-devel
+BuildRequires:  systemd-rpm-macros
+BuildRequires:  util-linux-devel
 # For tests
 BuildRequires:  iproute
+BuildRequires:  libcmocka-devel
+# Blocked: not building on 3.0 yet
+# BuildRequires:  openvswitch
+BuildRequires:  python%{python3_pkgversion}-cffi
 BuildRequires:  python%{python3_pkgversion}-coverage
 BuildRequires:  python%{python3_pkgversion}-netifaces
 BuildRequires:  python%{python3_pkgversion}-pycodestyle
+BuildRequires:  python%{python3_pkgversion}-pytest
+BuildRequires:  python%{python3_pkgversion}-pytest-cov
 BuildRequires:  python%{python3_pkgversion}-PyYAML
 
-# /usr/sbin/netplan is a Python 3 script that requires netifaces and PyYAML
-Requires:       python%{python3_pkgversion}-netifaces
-Requires:       python%{python3_pkgversion}-PyYAML
+# netplan ships dbus files and imports the dbus python module
+Requires:       dbus
+Requires:       python%{python3_pkgversion}-dbus
+
 # 'ip' command is used in netplan apply subcommand
 Requires:       iproute
 
-# netplan supports either systemd or NetworkManager as backends to configure the network
-Requires:       systemd
-Requires:       wpa_supplicant
+# /usr/sbin/netplan is a Python 3 script that requires netifaces, PyYAML,
+# cffi and optionally python-rich
+Requires:       python%{python3_pkgversion}-netifaces
+Requires:       python%{python3_pkgversion}-PyYAML
+Requires:       python%{python3_pkgversion}-cffi
+# Not available in 3.0 yet
```

Comment:
can this be merged without python3-rich?  or is this PR blocked?

Reply:
It can be merged. `rich` is an optional dependency, the code can handle it not being present. That said, we'll be adding it in the future.

Reply:
having a comment explaining that would be nice :)

Reply:
Done :)